cmd/dep/root_analyzer.go

```diff
@@ -53,6 +63,65 @@ func (a *rootAnalyzer) InitializeRootManifestAndLock(dir string, pr gps.ProjectR
 	return
 }
 
+func (a *rootAnalyzer) cacheDeps(pr gps.ProjectRoot) error {
+	deps := make(map[gps.ProjectRoot]bool)
+	logger := a.ctx.Err
+	g, ctx := errgroup.WithContext(context.TODO())
+	sem := make(chan struct{}, concurrency)
+
+	syncDep := func(pr gps.ProjectRoot, sm gps.SourceManager) error {
+		if err := sm.SyncSourceFor(gps.ProjectIdentifier{ProjectRoot: pr}); err != nil {
+			logger.Printf("Unable to cache %s - %s", pr, err)
+			return err
+		}
+		logger.Printf("Cached %s", pr)
+		return nil
+	}
+
+	for ip := range a.directDeps {
+		logger.Printf("Package %q, analyzing...", ip)
+		if paths.IsStandardImportPath(ip) {
+			continue
+		}
+		if hasImportPathPrefix(ip, string(pr)) {
+			continue
+		}
+
+		pr, err := a.sm.DeduceProjectRoot(ip)
```

Comment:
So removing this guy seems to be breaking the underlying call `SyncSourceFor` called by the new `syncDep` func. Im still investigating

Reply:
Ive been trying to use type conversion to pass a "gps.ProjectRoot" without the call to `DeduceProjectRoot`. 
err := syncDep(gps.ProjectRoot(ip), a.sm)

Doing this causes one of the deps to cache over and over again and then end the entire process. Digging in to this now.

```
Cached github.com/docker/spdystream
Cached github.com/fsnotify/fsnotify
Cached github.com/fsnotify/fsnotify
Cached github.com/xanzy/go-cloudstack
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Cached github.com/spf13/cobra
Successfully cached all deps.
```

Reply:
Odd, that is the right thing to do to cast to a ProjectRoot. I wonder if that's really due to other changes you've made to the code? 

I really wish that when I had refactored the importers that I had fixed the type on directDeps and used gps.ProjectRoot instead of string. 😀